Mathlib/Init/Algebra/Order.lean

```diff
@@ -383,6 +383,20 @@ theorem compare_trichotomy {a b : α} :
     (fun h ↦ Or.inr <| Or.inl (compare_eq_iff_eq.2 h))
     (fun h ↦ Or.inr <| Or.inr (compare_gt_iff_gt.2 h))
 
+-- TODO: generalize to instance of LE, Eq (and upstream?)
+-- Move?
+/-- The `Prop` corresponding to each `Ordering` constructor (e.g. `.lt.toProp a b` is `a < b`). -/
+def Ordering.toProp (o : Ordering) (a b : α) : Prop := match o with
+| .lt => a < b
+| .eq => a = b
+| .gt => a > b
```

Comment:
Perhaps cleaner as
```suggestion
def Ordering.toRel (o : Ordering) : α -> α -> Prop := match o with
| .lt => (· < ·)
| .eq => eq·
| .gt => (· > ·)
```
(but with unicode)

Reply:
Nice! I also absorbed `Ordering` into the type signature to get rid of the explicit `match` keyword. :)